hadoop-hdds/client/src/test/java/org/apache/hadoop/hdds/scm/client/TestHddsClientUtils.java

```diff
@@ -225,8 +232,38 @@ public void testVerifyResourceName() {
 
     for (String name : invalidNames) {
       assertThrows(IllegalArgumentException.class, () -> HddsClientUtils.verifyResourceName(name),
-          "Did not reject invalid string [" + name + "] as a name");
+          getInvalidNameMessage(name));
     }
+
+    // Message should include the name and resource type, if the name is of
+    // illegal length.
+    List<String> resourceTypes = ImmutableList.of("bucket", "volume");
+    for (int i = 0; i < 2; i++) {
+      String resType = resourceTypes.get(i);
+      String otherResType = resourceTypes.get(1 - i);
+      Throwable thrown = assertThrows(
+          IllegalArgumentException.class,
+          () -> HddsClientUtils.verifyResourceName(tooShort, resType, true),
+          getInvalidNameMessage(tooShort));
+
+      String message = thrown.getMessage();
+      assertNotNull(message);
+      assertTrue(message.contains(resType));
+      assertFalse(message.contains(otherResType));
+      assertTrue(thrown.getMessage().contains(tooShort));
+    }
+
+    // However, such names also containing unsupported characters should not be
+    // logged verbatim. This is to avoid vulnerabilities like Log4Shell.
+    final String tooShortInvalidChar = "$a";
+    Throwable thrown = assertThrows(
+        IllegalArgumentException.class,
+        () -> HddsClientUtils.verifyResourceName(tooShortInvalidChar),
+        getInvalidNameMessage(tooShortInvalidChar));
+
+    String message = thrown.getMessage();
+    assertNotNull(message);
+    assertFalse(message.contains(tooShortInvalidChar));
```

Comment:
Let's add these new test cases as separate two test methods.  I think `tooShort` can be moved, no need to test it twice.

Reply:
It crossed my mind the first time around. Sounds like I made the wrong call then. :slightly_smiling_face: Sure, it's split up now, thank you.